src/rsa.py

```diff
@@ -80,45 +80,57 @@ def sign(body, user):
         except Exception as ex:
             return {"error": f"Failed to base64-decode payload: {ex}"}, 400
 
-        tmpinput = tempfile.mkstemp(prefix="rsa-")[1]
-        with open(tmpinput, 'wb') as ifile:
-            ifile.write(payload)
-            tmpfile = tempfile.mkstemp(prefix="rsa-")[1]
-            cmd = [ "openssl", "dgst", "-sign",
-                   os.path.join(RSA_DIR, key_id, "private.pem"),
-                   "-keyform", "PEM",
-                   "-sha256", "-out", tmpfile, tmpinput ]
-            cmd_result = subprocess.run(cmd, check=False)
-            if cmd_result.returncode != 0:
-                raise RuntimeError("Failed to sign payload")
-            tmpout = tempfile.mkstemp(prefix="rsa-")[1]
-            with open(tmpout, 'a', encoding='utf-8') as output:
-                output.write(sha256sum(tmpinput))
-                output.write("\n")
-                output.write("ts: " +
+        with tempfile.NamedTemporaryFile(prefix="rsa-", delete=False,
+                                         mode='wb') as tmpinput:
+            tmpinput.write(payload)
+            tmpinput_path = tmpinput.name
+
+        try:
+            with tempfile.NamedTemporaryFile(prefix="rsa-", delete=False,
+                                             mode='wb') as tmpfile:
+                tmpfile_path = tmpfile.name
+
+                cmd = [ "openssl", "dgst", "-sign",
+                       os.path.join(RSA_DIR, key_id, "private.pem"),
+                       "-keyform", "PEM",
+                       "-sha256", "-out", tmpfile_path, tmpinput_path ]
+                subprocess.run(cmd, check=True)
+
+            with tempfile.NamedTemporaryFile(prefix="rsa-", delete=False,
+                                             mode='a', encoding='utf-8') as tmpout:
```

Comment:
Also nitpick - do we need a temp file here at all? It looks like the file is just written and then read back, no external process needs to use it, can we just build the string in memory?

Reply:
We probably could - it's probably just an artifact of translating bash into python too literally. As it is used to send the actual response back I don't think it's worth changing and retesting though. Let me know if you have a strong preference and I can do it.